### ISSUE TITLE
Make it easier to open other apps and files in a running DisposableVM

### ISSUE THREAD

**Issue by lattice0** [QubesOS/qubes-issues#5557]
**The problem you're addressing (if any)**
If I open a new DispVM, I cannot (easily) open other apps in the same DispVM. If I do through the qubes menu, it opens a new DispVM with the new App. 

**Describe the solution you'd like**
It'd be better if, when I opened, for example, Firefox in a DispVm called DispVM 3412, then DispVM 3412 appears in the Qubes menu just as other VMs do, so I can launch other apps other than Firefox inside it, not in a new one.

**Where is the value to a user, and who might that user be?**
In my case I want to test random github projects that require, for example, installing through terminal, browsing files, opening vscode, all inside one VM. Then I want everything to go away with the click of a button (shutdown VM). It's much better than creating a VM just for that and then erasing it. 
But I see people using this for any kind of problems

**Comment by marmarek** [comment]
You can open terminal using domains widget (Q icon). But generally using multiple applications in a single DisposableVM may be confusing, as DisposableVM is destroyed as soon as you close the "primary" application (the one started initially) - regardless of what else is running there.

**Comment by brendanhoar** [comment]
If I want to use a disposable VM for more than just a one-time task like opening a PDF of unknown providence (the intended purpose), I usually just start the disposableVM via the "Xfce Terminal" (whonix) or Xterm (fedora).

I then open another terminal window from that "original" terminal window and minimize it to reduce the chances of shutdown without intention.

Then use the secondary terminal window to execute commands/applications.

B

**Comment by ninavizz** [comment]
Ohboyy I did not know this is how things work today. Will give this some thought in the updated AppMenu design.

**Comment by deeplow** [comment]
There is a way to currently do this, but it takes some configuration. I should make a guide about it as now all that exists are a documentation page [on something related](https://github.com/Qubes-Community/Contents/blob/master/docs/common-tasks/opening-urls-in-vms.md) and a [discussion on the forum](https://qubes-os.discourse.group/t/opening-more-programs-in-running-disposable-appvm/2389).

The solution is to have a named dispVM and configuring the `qubes.OpenInVM` with something like:

```
srcVM $dispvm allow,target=namedDispVM
```

One thing the user would also need to do is to have the [qubes-app-shutdown-idle](https://www.qubes-os.org/doc/templates/minimal/#distro-specific-notes) package installed because now the qube doesn't shut down as soon as the user closes the viewing application (because there may be multiple).

---

## (slightly more) User-friendly implementation

In a qube's settings under `Advanced` tab » Other, there is the option `Default DisposableVM Template`. Currently only DisposableVM Templates there, but it would be nice to have the option to select Named DisposableVMs.

The user would still need to create said Named DisposableVM via the terminal and install the [qubes-app-shutdown-idle](https://www.qubes-os.org/doc/templates/minimal/#distro-specific-notes) package. But it would avoid having to mess with Qubes RPC configurations.

**Comment by SvenSemmler** [comment]
> the qube doesn't shut down as soon as the user closes the viewing
> application (because there may be multiple).

Are you sure?

If I start a DispVM with let's say Firefox and then I also open an 
XTerm. As soon as I close the Firefox window the qube shutsdown with 
XTerm and all.

It's easy to imagine a user losing hours of work that way if unaware.

**Comment by deeplow** [comment]
Yes, some thought will need to be given to the shut down behavior in a way that does not surprise the user. I mention [here](https://github.com/freedomofpress/securedrop-workstation/issues/333#issuecomment-791308579) another possible solution: shutting down the VM when all windows are closed.

**Comment by ninavizz** [comment]
This mockup is one of the leading candidate solutions for a new appmenu. It demonstrates the above disp-vm functionality. This feels like the most intuitive way to reflect this functionality in the appmenu. The ability to shut down VMs within the appmenu will likely be included in the forthcoming appmenu redesign (yep, that has been responded to quite favorably in the survey). Thoughts (on just the disp-vm part of this, relevant to this issue)?
![valentina-withopendisp](https://user-images.githubusercontent.com/8262612/117898150-25ce5c00-b279-11eb-8a2e-00462064dcb6.png)

**Comment by andrewdavidwong** [comment]
> It demonstrates the disp-vm functionality suggested in this issue.

Where? I'm not seeing any DisposableVMs in the menu.

**Comment by ninavizz** [comment]
@andrewdavidwong `datascrub` is the template for the disp-vm, and then `3573-datascrub` is the running disp-vm.

What are you "seeing" instead?

**Comment by andrewdavidwong** [comment]
> @andrewdavidwong `datascrub` is the template for the disp-vm, and then `3573-datascrub` is the running disp-vm.
> 
> 
> 
> What are you "seeing" instead? 

Oh, I was looking for an entry like `disp1234`, since that's how DisposableVMs are named in Qubes.

**Comment by DemiMarie** [comment]
> > @andrewdavidwong `datascrub` is the template for the disp-vm, and then `3573-datascrub` is the running disp-vm.
> > What are you "seeing" instead?
> 
> Oh, I was looking for an entry like `disp1234`, since that's how DisposableVMs are named in Qubes.

Nit: That is how automatically created DisposableVMs are named.  Manually-created DisposableVMs (often called “named DispVMs”) can be named anything the user wishes.  They also persist across reboots (both of the DispVM and of the host), although their data is wiped after shutdown.

**Comment by andrewdavidwong** [comment]
> > > @andrewdavidwong `datascrub` is the template for the disp-vm, and then `3573-datascrub` is the running disp-vm.
> > > What are you "seeing" instead?
> > 
> > 
> > Oh, I was looking for an entry like `disp1234`, since that's how DisposableVMs are named in Qubes.
> 
> Nit: That is how automatically created DisposableVMs are named. Manually-created DisposableVMs (often called “named DispVMs”) can be named anything the user wishes.

Sure, but then I would have expected something like, "In this screenshot, we've created a static DisposableVM named X." At first, I thought `creo` was the DisposableVM, since that's the one with the open menu. But then I thought, "Wait, there's no indication that that's a DisposableVM, so that must not be it. But then where is it?" which led to my question.

So, one important takeaway is to indicate to the user which of these VMs are disposable and which are not.

**Comment by deeplow** [comment]
> This feels like the most intuitive way to reflect this functionality in the appmenu.

Quite an interesting approach. I like it :+1:

**Comment by ninavizz** [comment]
Alrightee... it sounds like there are some conversations that need to happen, around all the incarnation options for disposable VMs. That can happen, and I'll put it on my "to do" list for an async workshop. 

One other issue, is that the new icons system has not yet rolled-out... and I don't even have `4.1` on a laptop, to "test" how Marta has implemented the disp-vm icon. I also don't know if the disp-vm's are still currently italicized in the menu, or no. 

I will work on procuring a laptop to run test builds on, as I probably should have that, anyway. I cannot unfortunately run 4.1 on my SecureDrop Workstation laptop, which thus far is my only Qubes machine.

**Comment by ninavizz** [comment]
Functionality shown in #6665

**Comment by deeplow** [comment]
Implementing this will possibly have to change behavior of when a disposable qube shuts down.

## Current Behavior

Shut down as soon as the first program opened there is closed. Example:

1. opens pdf in `disp1234`
2. opens browser in `disp1234`
3. closes pdf reader and qube shuts down (including the still open browser)

## Future Behavior

Shuts down only when all the processes associated with open applications close.

1. opens pdf in `disp1234`
2. opens browser in `disp1234`
3. closes pdf reader
4. closes browser
5. qube shuts down

**Comment by ninavizz** [comment]
@marmarta Has this been done or is it being worked on?

**Comment by marmarta** [comment]
this is done in appmenu for a while now (tied to first app open)
